components/resourceDetails/resourceSharing/agentAccessTable/index.jsx

```diff
@@ -27,30 +27,32 @@ import PropTypes from "prop-types";
 import { useFilters, useGlobalFilter, useTable } from "react-table";
 import { useBem } from "@solid/lit-prism-patterns";
 import clsx from "clsx";
-import {
-  Accordion,
-  createStyles,
-  IconButton,
-  InputBase,
-  Tab,
-  Tabs,
-  Typography,
-} from "@material-ui/core";
+import { Accordion, createStyles, Typography } from "@material-ui/core";
 import { makeStyles } from "@material-ui/styles";
+import useNamedPolicyPermissions from "../../../../src/hooks/useNamedPolicyPermissions";
+import usePermissionsWithProfiles from "../../../../src/hooks/usePermissionsWithProfiles";
 import AgentAccess from "../agentAccess";
 import AddAgentButton from "../addAgentButton";
+import AgentsTableTabs from "../agentsTableTabs";
+import AgentsSearchBar from "../agentsSearchBar";
 
 import styles from "./styles";
+import PolicyHeader from "../policyHeader";
 
 const useStyles = makeStyles((theme) => createStyles(styles(theme)));
 const TESTCAFE_ID_SHOW_ALL_BUTTON = "show-all-button";
 const TESTCAFE_ID_HIDE_BUTTON = "hide-button";
-const TESTCAFE_ID_SEARCH_INPUT = "search-input";
-const TESTCAFE_ID_TAB_ALL = "tab-all";
-const TESTCAFE_ID_TAB_PEOPLE = "tab-people";
-const TESTCAFE_ID_TAB_GROUPS = "tab-groups";
 
-export default function AgentAccessTable({ permissions, type }) {
+export default function AgentAccessTable({ type }) {
+  const {
+    data: namedPermissions,
+    mutate: mutatePermissions,
+  } = useNamedPolicyPermissions(type);
+
+  const { permissionsWithProfiles: permissions } = usePermissionsWithProfiles(
```

Comment:
`usePermissionsWithProfiles` should be rewritten into only doing one permission at a time, which should be called from the components that deal with only one permission at a time.

Reply:
That doesn't make sense here, because the permissions are being memoized and used in a table, and because the length is used in a few places. Typically, yes, you would want to separate it out so that sub-components would each use permissions themselves.

Reply:
Also we do need the profiles in the table beforehand to do things like filtering and sorting.